docs/src/charm/howto/upgrade-minor.md

```diff
@@ -0,0 +1,210 @@
+* the kubernetes channel this charm follows (ex. `1.31/stable`)
+* the current charm revision  (ex. `1001`)
+* the next potential charm revision (ex. `ch:amd64/k8s-1002`)
+
+If the `can-upgrade-to` revision is `null`, you are at the most
+stable release in this channel and you should continue with the
+[Pre Upgrade Check](#the-pre-upgrade-check).
+
+If the `can-upgrade-to` revision is non-null, continue with the
+[Upgrade Patch](upgrade-patch) instructions.
+
+
+```{caution} Only update the charm to the next minor version.
+If the current `charm-channel` is `1.31/stable`, it's critical
+to refresh to the `1.32/stable`. Skipping channels (eg 1.31 -> 1.33)
+will result in the units blocking and indicating they cannot upgrade.
+```
+
+### The pre-upgrade-check
+
+Before running an upgrade, we should check that the cluster is 
+steady and ready for upgrade. The charm will perform checks 
+necessary to confirm the cluster is in safe working order before
+upgrading.
+
+```sh
+juju run k8s/leader pre-upgrade-check
+```
+
+If no error appears, the `pre-upgrade-check` completed successfully.
+
+### Refreshing charm applications
+
+#### Control Plane units (k8s)
+
+Following the `pre-upgrade-check` update the control-plane nodes.
+
+```sh
+juju refresh k8s --channel ${NEXT_CHANNEL}
+juju status k8s --watch 5s
+```
+
+The `refresh` command instructs the juju controller to follow a new
+charm channel related to the Kubernetes release and use the new charm
+revision of the application's channel to upgrade each unit. The
+charm code is simultaneously replaced on each unit, then the `k8s`
+snap is updated unit-by-unit, starting with the Juju leader unit for the
+application.
+During the upgrade process, the application status message and the `k8s` leader
+unit message will display the current progress, listing the `k8s` and
+`k8s-worker` units still pending upgrades.
+After the `k8s` charm is upgraded, the application `Version` from `juju status`
+will reflect the updated version of the control-plane nodes making up the cluster.
+
+#### Worker units (k8s-worker)
+
+After updating the control-plane applications, worker nodes may be upgraded
+following running the `pre-upgrade-check`. 
+
+```sh
+juju run k8s-worker/leader pre-upgrade-check
+juju refresh k8s-worker --channel ${NEXT_CHANNEL}
+juju status k8s-worker --watch 5s
+```
+
+The `refresh` command instructs the juju controller to follow a new
+charm channel related to the Kubernetes release and use the new charm
+revision of the application's channel to upgrade each unit. The
+charm code is simultaneously replaced on each unit, then the `k8s`
+snap is updated unit-by-unit, starting with the Juju leader unit for the
+application.
+
+After the `k8s-worker` charm is upgraded, the application `Version` from `juju status`
+will reflect the updated version of the worker nodes making up the cluster.
+
+```{note} Repeat for every application using the k8s-worker charm if
```

Comment:
Can we be more precise here please? What are we repeating and what appears multiple times?